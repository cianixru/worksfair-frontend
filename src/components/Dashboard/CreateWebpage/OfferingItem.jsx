import React from 'react';
import PropTypes from 'prop-types';

const OfferingItem = ({
  offerings,
}) => {
  return (
    <div>
      { offerings.length > 0
        && offerings.map(offering => (
          <div
            className="columns card card-content"
            key={offering.title}
          >
            <div className="column is-4">
              <img
                src={offering.image}
                alt=""
                className="offering-image"
              />
            </div>
            <div className="column is-7">
              <h4 className="title is-5 has-text-grey-darker">
                {offering.title}
              </h4>
              <p>{offering.description}</p>
            </div>
            <div className="column is-1">
              <p>{offering.price}</p>
            </div>
          </div>
        ))
      }
    </div>
  );
};

OfferingItem.propTypes = {
  offerings: PropTypes.array,
};

export default OfferingItem;
