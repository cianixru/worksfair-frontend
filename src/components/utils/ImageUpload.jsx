import React from 'react';
import Dropzone from 'react-dropzone';
import PropTypes from 'prop-types';

import sampleImage from '../../assets/sample-02.jpg';

const ImageUpload = ({
  selectedImage,
  handleOfferingImageSelection,
}) => {
  /**
   * @description Handles the selection of an image
   *
   * @param { Array } files
   */
  const handleImageSelection = (files) => {
    handleOfferingImageSelection(files);
  };

  return (
    <div className="image-upload-box">
      <img
        src={selectedImage || sampleImage}
        alt=""
        className="offering-image" />
      <div className="change-image-btn">
        <Dropzone
          name="image"
          // eslint-disable-next-line react/jsx-no-bind
          onDrop={handleImageSelection}
          accept="image/*"
          multiple={false}
        >
          {({ getRootProps, getInputProps }) => (
            <div {...getRootProps()} className="imageUpload">
              <span><i className="fa fa-upload" /> Upload Pic</span>
              <input {...getInputProps()} />
            </div>
          )}
        </Dropzone>
      </div>
    </div>
  );
};

ImageUpload.propTypes = {
  handleOfferingImageSelection: PropTypes.func,
  selectedImage: PropTypes.string,
};

export default ImageUpload;
