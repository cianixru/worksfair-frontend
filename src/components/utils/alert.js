import { toast, Slide } from 'react-toastify';

class notify {
  static success(message) {
    toast.success(message, {
      hideProgressBar: false,
      transition: Slide,
      autoClose: 4000,
      pauseOnHover: true,
      draggable: true,
    });
  }

  static error(message) {
    toast.error(message, {
      hideProgressBar: false,
      transition: Slide,
      autoClose: 4000,
      pauseOnHover: true,
      draggable: true,
    });
  }

  static info(message) {
    toast.info(message, {
      hideProgressBar: false,
      transition: Slide,
      autoClose: false,
      draggable: true,
    });
  }

  static warning(message) {
    toast.warn(message, {
      hideProgressBar: false,
      transition: Slide,
      autoClose: 4000,
      pauseOnHover: true,
      draggable: true,
    });
  }
}

export default notify;
