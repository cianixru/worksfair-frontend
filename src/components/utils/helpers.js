import React from 'react';
import axios from 'axios';

const {
  REACT_APP_CLOUDINARY_API_KEY,
  REACT_APP_CLOUDINARY_PRESET_NAME,
  REACT_APP_CLOUDINARY_UPLOAD_URL,
} = process.env;

/**
 * @description convert a sentence to sentence case
 *
 * @param { string } sentence
 * @returns {string }
 */
export const toSentenceCase = sentence => sentence
    && `${sentence[0].toUpperCase()}${sentence.substr(1)}`;

/**
 * @description Handles the image upload to cloudinary
 *
 * @param { object } image
 */
export const CloudinaryImageUploader = async (image) => {
  // our formdata
  const formData = new FormData();
  formData.append('file', image);
  formData.append(
    'api_key',
    REACT_APP_CLOUDINARY_API_KEY
  );
  formData.append(
    'upload_preset',
    REACT_APP_CLOUDINARY_PRESET_NAME
  );
  formData.append('timestamp', (Date.now()));

  delete axios.defaults.headers.common.Authorization;

  return axios({
    method: 'post',
    url: REACT_APP_CLOUDINARY_UPLOAD_URL,
    data: formData,
    config: { headers: { 'Content-Type': 'multipart/form-data' } }
  });
};

export const offerings = [
  {
    title: 'kamsi',
    description: 'passion, software, engineering, technology, worksfair',
    image: 'https://res.cloudinary.com/worksfair/image/upload/v1557767623/dev/Andela-VI_gfqg0d.jpg',
    price: 'FREE',
    created_at: '2019-05-15T17:37:06.479115Z',
    updated_at: '2019-05-15T17:37:06.479140Z'
  },
  {
    title: 'Sissy',
    description: 'passion, software, engineering, technology, worksfair',
    image: 'https://res.cloudinary.com/worksfair/image/upload/v1557767623/dev/Andela-VI_gfqg0d.jpg',
    price: '',
    created_at: '2019-05-15T17:35:00.329484Z',
    updated_at: '2019-05-15T17:35:00.329604Z'
  },
  {
    title: 'Atty',
    description: 'worksfair will pay for you',
    image: 'https://res.cloudinary.com/worksfair/image/upload/v1557767623/dev/Andela-VI_gfqg0d.jpg',
    price: 'FREE',
    created_at: '2019-05-15T17:29:04.245897Z',
    updated_at: '2019-05-15T17:40:32.493396Z'
  },
  {
    title: 'Cheesy',
    description: null,
    image: 'https://res.cloudinary.com/worksfair/image/upload/v1557767623/dev/Andela-VI_gfqg0d.jpg',
    price: '',
    created_at: '2019-05-15T05:49:37.499647Z',
    updated_at: '2019-05-15T05:49:37.499668Z'
  }
];

export const addNaira = (price) => {
  const cost = parseInt(price, 10) === 0
    ? <p className="is-size-6">Contact us for Price</p>
    : (<p className="price">&#8358; {price}</p>);
  return cost;
};
