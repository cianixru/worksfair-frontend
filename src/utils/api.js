import axios from 'axios';

require('dotenv').config({ path: '../../.env' });

export const baseURL = process.env.REACT_APP_API_URL;
const api = axios.create({
  baseURL,
});

export default api;
