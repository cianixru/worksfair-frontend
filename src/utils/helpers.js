/**
 * @description formats the error response messages
 * @param { array } data
 * @returns { string } errorMessage
 */
export default (data) => {
  const errors = Object.values(data);
  const errorMessage = errors.reduce((error, arr) => {
    return `${error} ${arr[0]}`;
  }, '');
  return errorMessage;
};
