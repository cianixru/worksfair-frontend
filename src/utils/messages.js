// eslint-disable-next-line import/prefer-default-export
export const tooltipMessage = {
  businessName: 'Keep it short. It is also your Webpage title.',
  keywords: 'Some search keywords relating to your business.'
    + 'It will help people to find your business.',
  description: 'Your customers will see this message on webpage.',
  colour: 'The colour theme for your webpage. the default is plain gray-scale'
};
