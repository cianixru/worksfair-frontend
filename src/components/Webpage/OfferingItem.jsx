import React from 'react';
import PropTypes from 'prop-types';

import { addNairaSign } from '../utils/helpers';

const OfferingItem = ({
  offerings,
  colour,
}) => {
  return (
    <ul className="columns">
      { offerings.length > 0
        && offerings.map(offering => (
          <div
            key={offering.title}
            className="column is-6">
            <li
              className="box"
            >
              <div className="columns is-desktop">
                <div className="column">
                  <figure className="image is-180x180">
                    <img
                      src={offering.image}
                      alt=""
                      className="offering-image"
                    />
                  </figure>
                </div>
                <div className="column">
                  <h4 className="title is-5">
                    {offering.title}
                  </h4>
                  <p>{offering.description}</p>
                  <div className={`tag ${colour} is-large`}>
                    {addNairaSign(offering.price)}
                  </div>
                </div>
              </div>
            </li>
          </div>
        ))
      }
    </ul>
  );
};

OfferingItem.propTypes = {
  offerings: PropTypes.array,
};

export default OfferingItem;
