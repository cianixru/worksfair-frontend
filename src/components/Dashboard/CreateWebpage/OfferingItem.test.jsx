import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup } from 'react-testing-library';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';

import OfferingItem from './OfferingItem';
import { offerings } from '../../utils/helpers';

afterEach(cleanup);

const middlewares = [thunk];
const store = configureMockStore(middlewares);
const mockStore = store({
  auth: { currentUser: {} },
  webpage: { newWebpage: {} }
});

describe('OfferingItem.jsx', () => {
  const onSubmit = jest.fn();
  const user = {};
  const validationErrors = {};
  const handleErrorReset = jest.fn();

  test('that component renders correctly', async () => {
    const component = render(
      <Provider store={mockStore}>
        <Router>
          <OfferingItem
            onSubmit={onSubmit}
            user={user}
            validationErrors={validationErrors}
            handleErrorReset={handleErrorReset}
            offerings={offerings}
          />
        </Router>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
