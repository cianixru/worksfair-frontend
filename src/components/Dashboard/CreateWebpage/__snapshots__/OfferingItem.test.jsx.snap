// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferingItem.jsx that component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="columns card card-content"
        >
          <div
            class="column is-4"
          >
            <img
              alt=""
              class="offering-image"
              src="https://res.cloudinary.com/worksfair/image/upload/v1557767623/dev/Andela-VI_gfqg0d.jpg"
            />
          </div>
          <div
            class="column is-7"
          >
            <h4
              class="title is-5 has-text-grey-darker"
            >
              kamsi
            </h4>
            <p>
              passion, software, engineering, technology, worksfair
            </p>
          </div>
          <div
            class="column is-1"
          >
            <p>
              FREE
            </p>
          </div>
        </div>
        <div
          class="columns card card-content"
        >
          <div
            class="column is-4"
          >
            <img
              alt=""
              class="offering-image"
              src="https://res.cloudinary.com/worksfair/image/upload/v1557767623/dev/Andela-VI_gfqg0d.jpg"
            />
          </div>
          <div
            class="column is-7"
          >
            <h4
              class="title is-5 has-text-grey-darker"
            >
              Sissy
            </h4>
            <p>
              passion, software, engineering, technology, worksfair
            </p>
          </div>
          <div
            class="column is-1"
          >
            <p />
          </div>
        </div>
        <div
          class="columns card card-content"
        >
          <div
            class="column is-4"
          >
            <img
              alt=""
              class="offering-image"
              src="https://res.cloudinary.com/worksfair/image/upload/v1557767623/dev/Andela-VI_gfqg0d.jpg"
            />
          </div>
          <div
            class="column is-7"
          >
            <h4
              class="title is-5 has-text-grey-darker"
            >
              Atty
            </h4>
            <p>
              worksfair will pay for you
            </p>
          </div>
          <div
            class="column is-1"
          >
            <p>
              FREE
            </p>
          </div>
        </div>
        <div
          class="columns card card-content"
        >
          <div
            class="column is-4"
          >
            <img
              alt=""
              class="offering-image"
              src="https://res.cloudinary.com/worksfair/image/upload/v1557767623/dev/Andela-VI_gfqg0d.jpg"
            />
          </div>
          <div
            class="column is-7"
          >
            <h4
              class="title is-5 has-text-grey-darker"
            >
              Cheesy
            </h4>
            <p />
          </div>
          <div
            class="column is-1"
          >
            <p />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="columns card card-content"
      >
        <div
          class="column is-4"
        >
          <img
            alt=""
            class="offering-image"
            src="https://res.cloudinary.com/worksfair/image/upload/v1557767623/dev/Andela-VI_gfqg0d.jpg"
          />
        </div>
        <div
          class="column is-7"
        >
          <h4
            class="title is-5 has-text-grey-darker"
          >
            kamsi
          </h4>
          <p>
            passion, software, engineering, technology, worksfair
          </p>
        </div>
        <div
          class="column is-1"
        >
          <p>
            FREE
          </p>
        </div>
      </div>
      <div
        class="columns card card-content"
      >
        <div
          class="column is-4"
        >
          <img
            alt=""
            class="offering-image"
            src="https://res.cloudinary.com/worksfair/image/upload/v1557767623/dev/Andela-VI_gfqg0d.jpg"
          />
        </div>
        <div
          class="column is-7"
        >
          <h4
            class="title is-5 has-text-grey-darker"
          >
            Sissy
          </h4>
          <p>
            passion, software, engineering, technology, worksfair
          </p>
        </div>
        <div
          class="column is-1"
        >
          <p />
        </div>
      </div>
      <div
        class="columns card card-content"
      >
        <div
          class="column is-4"
        >
          <img
            alt=""
            class="offering-image"
            src="https://res.cloudinary.com/worksfair/image/upload/v1557767623/dev/Andela-VI_gfqg0d.jpg"
          />
        </div>
        <div
          class="column is-7"
        >
          <h4
            class="title is-5 has-text-grey-darker"
          >
            Atty
          </h4>
          <p>
            worksfair will pay for you
          </p>
        </div>
        <div
          class="column is-1"
        >
          <p>
            FREE
          </p>
        </div>
      </div>
      <div
        class="columns card card-content"
      >
        <div
          class="column is-4"
        >
          <img
            alt=""
            class="offering-image"
            src="https://res.cloudinary.com/worksfair/image/upload/v1557767623/dev/Andela-VI_gfqg0d.jpg"
          />
        </div>
        <div
          class="column is-7"
        >
          <h4
            class="title is-5 has-text-grey-darker"
          >
            Cheesy
          </h4>
          <p />
        </div>
        <div
          class="column is-1"
        >
          <p />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
